tests/web_api/detect_test.py

```diff
@@ -0,0 +1,39 @@
+import requests
+import unittest
+import importlib
+utils = importlib.import_module(
+    'extensions.sd-webui-controlnet.tests.utils', 'utils')
+utils.setup_test_env()
+
+
+class TestDetectEndpointWorking(unittest.TestCase):
+    def setUp(self):
+        self.base_detect_args = {
+            "controlnet_module": "canny",
+            "controlnet_input_images": [utils.readImage("test/test_files/img2img_basic.png")],
+            "controlnet_processor_res": 512,
+            "controlnet_threshold_a": 0,
+            "controlnet_threshold_b": 0,
+        }
+
+    def test_detect_with_invalid_module_performed(self):
+        detect_args = self.base_detect_args | {
```

Comment:
Although this is a good solution for python 3.10, this makes the test not work under python 3.8 as the dict union operator was introduced in python 3.9. It was discussed earlier that the repo should support both 3.8 and 3.10.

As this is code that will only be run during automated testing, this may be okay. However, this means we now can only run the tests using 3.10 now.

Reply:
I wasn't aware the repo should support 3.8 as well.
I pushed another update to cover it.
It would be nice if there's a contribution guide, maybe it's there already and I missed.

Reply:
No problem, thanks for your time updating the code.